### PR TITLE
Improve the dbg! macro docs 

### LINKED ISSUE #58383: Suppress dbg! in release build
I'm hesitant to open this issue, because I suspect it has an obvious answer that I'm missing.

Anyway...

I **love** the new dbg! macro in 1.32.0.  But, given that its purpose is outputting debugging messages, should its output be suppressed in a project's release build?  (cargo build --release)

### PR DESCRIPTION
**Pull request by DevQps**
# Description

As stated has been discussed in #58383 the docs do not clearly state why it is useful to have the option to use `dbg!` in release builds as well. This PR should change that.

closes #58383

### CODE REVIEW

File: src/libstd/macros.rs [hunk 1]
```diff
@@ -231,10 +231,13 @@ macro_rules! eprintln {
 /// to give up ownership, you can instead borrow with `dbg!(&expr)`
 /// for some expression `expr`.
 ///
+/// The `dbg!` macro works exactly the same in release builds. This is useful when debugging issues
```
Comment (Centril) [inline src/libstd/macros.rs]: The textual changes here are good but the indentation is weird as compared with the rest of the text... can you adjust a bit? E.g. split the line after the first `.`

Comment (DevQps) [reply src/libstd/macros.rs]: Done!

Comment (Centril) [reply src/libstd/macros.rs]: I don't think there's any standard. It's mostly to make it work nicely with the rest of the text.

File: src/libstd/macros.rs [hunk 2]
```diff
@@ -231,10 +231,14 @@ macro_rules! eprintln {
 /// to give up ownership, you can instead borrow with `dbg!(&expr)`
 /// for some expression `expr`.
 ///
+/// The `dbg!` macro works exactly the same in release builds.
+/// This is useful when debugging issues that only occur in release builds or when debugging in
```
Comment (Centril) [inline src/libstd/macros.rs]:
```suggestion
/// This is useful when debugging issues that only occur in release
/// builds or when debugging in release mode is significantly faster.
```

File: src/libstd/macros.rs [hunk 3]
```diff
@@ -231,10 +231,14 @@ macro_rules! eprintln {
...
+/// release mode is significantly faster.
```
Comment (Centril) [inline src/libstd/macros.rs]:
```suggestion
```

**Comment by DevQps** [comment]
@Centril Thanks for the suggestions. I wasn't aware that the text was written up to a specific length (makes sense now I look at it). I committed your changes!

**Comment by Centril** [comment]
Thanks! @bors r+ rollup

File: src/libstd/macros.rs (hunk 1 above)
Comment (steveklabnik) [reply src/libstd/macros.rs]: There is a standard: https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text

Comment (steveklabnik) [reply src/libstd/macros.rs]: (That said, this does not list line lengths, but the standard library wraps to 80 columns as its standard style)

Comment (DevQps) [reply src/libstd/macros.rs]: @steveklabnik Ahh great! I'll give it a read!

File: src/libstd/macros.rs [hunk 4]
```diff
@@ -306,6 +310,8 @@ macro_rules! eprintln {
 /// ```
 ///
 /// [stderr]: https://en.wikipedia.org/wiki/Standard_streams#Standard_error_(stderr)
+/// [debug-log]: https://docs.rs/log/*/log/macro.debug.html
+/// [log]: https://docs.rs/log/
```
Comment (lzutao) [inline src/libstd/macros.rs]:
```diff
-/// [debug-log]: https://docs.rs/log/*/log/macro.debug.html
-/// [log]: https://docs.rs/log/
+/// [`debug!`]: https://docs.rs/log/*/log/macro.debug.html
+/// [`log`]: https://crates.io/crates/log
```

File: src/libstd/macros.rs [hunk 5]
```diff
@@ -231,10 +231,14 @@ macro_rules! eprintln {
...
 ///
+/// The `dbg!` macro works exactly the same in release builds.
+/// This is useful when debugging issues that only occur in release
+/// builds or when debugging in release mode is significantly faster.
+///
 /// Note that the macro is intended as a debugging tool and therefore you
 /// should avoid having uses of it in version control for longer periods.
 /// Use cases involving debug output that should be added to version control
-/// may be better served by macros such as `debug!` from the `log` crate.
+/// are better served by macros such as [`debug!`][debug-log] from the [`log`][log] crate.
```
Comment (lzutao) [inline src/libstd/macros.rs]:
```diff
-/// are better served by macros such as [`debug!`][debug-log] from the [`log`][log] crate.
+/// are better served by macros such as [`debug!`] from the [`log`] crate.
```

Comment (DevQps) [reply src/libstd/macros.rs]: Fixed this! I keep learning new things everyday... I had no idea you could do this without the brackets 👍

Comment (Centril) [reply src/libstd/macros.rs]: Please fix it in a follow up PR; your PR is rolled up and is being tested right now.

**Comment by Centril** [comment]
@DevQps Please cherry-pick https://github.com/rust-lang/rust/pull/59528/commits/4c5ef86cf5ed137912ff5181e3807b840efc6bf8 onto a new PR instead. Also, we prefer not to have merge commits in our history.

**Comment by DevQps** [comment]
@Centril I am just trying to learn here a bit, so I hope you don't mind this question: But I seems that by doing the roll-up commit, you pushed the changes into master? Does that mean this branch can be closed now? I did some changes that @lzutao did as well, not sure if they were taken with the roll-up as well? Or am I just misunderstanding the whole roll-up part? :)

**Comment by Centril** [comment]
https://github.com/rust-lang/rust/pull/59550 hasn't been merged yet; but when it will, ordinarily this PR would also be merged. However, since you added commits it won't be automatically closed by GH. Instead, once the rollup is merged, close this PR manually and you can make a new PR with the other changes if you like.

**Comment by DevQps** [comment]
@Centril Alright thanks! I will close the other one I opened in meanwhile then, and just wait for this to finish and then add a new one! Thanks for all the support, stuff is a bit new here, so it takes me some time understand the complete process :)

**Comment by Centril** [comment]
https://github.com/rust-lang/rust/pull/59550 was merged.

**Pull request closed without merging**